kubera/authentication/authentication_test.go

```diff
@@ -0,0 +1,20 @@
+package authentication
+
+import (
+	"testing"
+
+	conn "github.com/mayadata-io/kubera-api-testing/kubera/connection"
+)
+
+func TestFetchToken(t *testing.T) {
+
```

Comment:
```go
func TestFetchToken(t *testing.T) {
 connection := conn.Get()
 var tests = map[string]struct{
   Hostname string
   Username string
   Password string  
   IsError bool
 }{
  "No host": {
     Username: connection.Username,
     Password: connection.Password,
     IsError: true,
   },
  "No username": {
     Hostname: connection.Hostname,
     Password: connection.Password,
     IsError: true,
  },
  "No password": {
     Hostname: connection.Hostname,
     Username: connection.Username,
     IsError: true,
  },
  "With host, username & password": {
     Hostname: connection.Hostname,
     Username: connection.Username,
     Password: connection.Password,
  },
 }
 for name, mock := range tests {
   name := name
   mock := mock
   t.Run(name, func(t *testing.T){
       fetcher := NewTokenFetcher(TokenFetchingConfig{
         Connection: Connection{
            Hostname: mock.Hostname,
            Username: mock.Username,
            Password: mock.Password,
         },
	})
	tkn, err := fetcher.Fetch()
	if mock.IsError && err.Err == nil {
          t.Fatalf("Expected error got none: %s", fetcher)
	}
	if !mock.IsError && err.Err != nil {
          t.Fatalf("Expected none error got %s: %s", err, fetcher)
	}
   })
 }
}
```

Reply:
@AmitKumarDas what is `mock` here ? Where to use and how ?

Reply:
@rohan2794  
In GoLang range loop syntax will be , 
```go
for index, item := range arrayItems {
}
```
while using MAPS iteration ,
 **The first value returned in map is key and the second value is the value of the key-value pair in map.**